frontend/src/concepts/pipelines/apiHooks/useArtifactStorage.ts

```diff
@@ -0,0 +1,68 @@
+import { useEffect, useState } from 'react';
+
+import { SupportedArea, useIsAreaAvailable } from '~/concepts/areas';
+import { fetchStorageObject, fetchStorageObjectSize } from '~/services/storageService';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+import { Artifact } from '~/third_party/mlmd';
+import {
+  extractS3UriComponents,
+  getArtifactUrlFromUri,
+} from '~/concepts/pipelines/content/artifacts/utils';
+import { useGetArtifactById } from './useGetArtifactById';
+
+export const useArtifactStorage = (
+  artifact?: Artifact,
+): {
+  artifactFeatureFlagEnabled: boolean;
+  objectUrl: string | undefined;
+  objectSize: number | undefined;
+  downloadedObject: string | undefined;
+} => {
+  const s3EndpointAvailable = useIsAreaAvailable(SupportedArea.S3_ENDPOINT).status;
+  const artifactApiAvailable = useIsAreaAvailable(SupportedArea.ARTIFACT_API).status;
+  const { namespace } = usePipelinesAPI();
+
+  const path = artifact?.getUri();
+  const artifactId = artifact?.getId().toString();
+
+  const [downloadedObject, setDownloadedObject] = useState<string | undefined>(undefined);
+  const [objectUrl, setStorageUrl] = useState<string | undefined>(undefined);
+  const [objectSize, setObjectSize] = useState<number | undefined>(undefined);
+  const artifactFeatureFlagEnabled = s3EndpointAvailable || artifactApiAvailable;
+  const [artifactResponse] = useGetArtifactById(artifactId, 'DOWNLOAD');
+  const artifactSize = Number(artifactResponse?.artifact_size);
+
+  useEffect(() => {
+    if (artifactApiAvailable) {
+      if (artifactResponse?.download_url) {
+        fetch(artifactResponse.download_url, {
+          headers: { 'Content-Type': 'text/plain' },
```

Comment:
@DaoDaoNoCode @Gkrumbach07 I am getting here `CORS` while fetching the download_url

Reply:
I am not seeing that on my end. how are you testing it

Reply:
@Gkrumbach07 , let me correct my statement
while fetching `download_url`, I am getting `cors` error 

<img width="1009" alt="Screenshot 2024-07-22 at 6 02 32 PM" src="https://github.com/user-attachments/assets/aa800be0-12bc-49ea-9c7e-f49e1a56cf03">

Reply:
I did this and it worked

```tsx
const [artifactResponse, loaded, error] = useGetArtifactById(artifact.getId(), 'DOWNLOAD');
  console.log(artifactResponse, loaded, error);

  useEffect(() => {
    if (artifactResponse?.download_url) {
      fetch(artifactResponse.download_url, {
        headers: { 'Content-Type': 'text/plain' },
      })
        .then((response) => response.text())
        .then((downloadObject) => {
          console.log(downloadObject);
        })
        .catch((e) => console.log(e));
    }
  }, [artifactResponse, namespace]);
```

I can try again with the useArtifactStorage. or if you have a testing commit i can look at